src/reducer.ts

```diff
@@ -108,50 +139,120 @@ type ReducerCreator = {
     actionCreator: UnknownActionCreator[],
     reducer: Reducer<State, VoidPayload>
   ): RegisteredReducer<State, VoidPayload>;
+
+  <State, Payload>(
+    source$: Observable<Payload>,
+    reducer: Reducer<State, Payload>
+  ): RegisteredReducer<State, Payload>;
 };
 
 export const reducer: ReducerCreator = <State>(
-  actionCreator: UnknownActionCreator | UnknownActionCreator[],
+  actionCreator:
+    | UnknownActionCreator
+    | UnknownActionCreator[]
+    | Observable<any>,
   reducerFn: Reducer<State, any>
-): RegisteredReducer<State, unknown> => {
+) => {
   const wrapper = (state: State, payload: any) => reducerFn(state, payload);
-  wrapper.trigger = {
-    actions: Array.isArray(actionCreator) ? actionCreator : [actionCreator],
-  };
+  if (actionCreator instanceof Observable) {
+    wrapper.trigger = {
+      source$: actionCreator,
+    };
+  } else {
+    wrapper.trigger = {
+      actions: wrapInArray(actionCreator),
+    };
+  }
   return wrapper;
 };
 
+const ACTION_ORIGIN = Symbol('Action origin');
+const SEED_ORIGIN = Symbol('Seed origin');
+
 /**
- * Combine reducer entries into a stream operator
+ * Combine registered reducers into a stream operator
  *
- * The payload of each incoming action is applied to the matching reducers
- * together with the previous state (or the seed if it's the first invocation),
- * and the returned state is emitted.
+ * Each reducer will receive the previous state (or the seed if it's the first
+ * invocation) together with the payloads of the actions of the given reducer,
+ * or the emitted values from the stream of the given reducer.
  *
- * This operator does not change whether the stream is hot or cold.
+ * The behaviour is undefined if multiple reducers are registered for the same
+ * actions.
+ *
+ * This operator will only emit on relevant actions or if one of the source
+ * streams emits. Specifically, if the seed is a stream, this operator will not
+ * emit when the seed stream emits, but the first time an action of source
+ * stream emits after the seed stream has emitted. The seed stream will be
+ * unsubscribed after its first emit.
```

Comment:
At the moment I don't se the use case of seed as stream, do we already use this in our app?

This 

> Specifically, if the seed is a stream, this operator will not
   emit when the seed stream emits, but the first time an action of source
   stream emits after the seed stream has emitted.

Seems not entirely trivial, what is the motivation?